src/app/Settings/DatetimeControl.tsx

```diff
@@ -0,0 +1,166 @@
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+
+  const [dateLocaleOpen, setDateLocaleOpen] = React.useState(false);
+  const [state, setState] = React.useState<DatetimeFormat>(defaultDatetimeFormat);
+
+  React.useLayoutEffect(() => {
+    context.settings.datetimeFormat().subscribe(setState);
+  }, [addSubscription, setState, context.settings]);
+
+  const handleDateToggle = React.useCallback((expanded: boolean) => setDateLocaleOpen(expanded), [setDateLocaleOpen]);
+
+  const handleDateLocaleSelect = React.useCallback(
+    (_, locale) => {
+      setDateLocaleOpen(false);
+      context.settings.setDatetimeFormat({
+        ...state,
+        dateLocale: {
+          name: locale.name,
+          key: locale.key,
+        },
+      });
+    },
+    [context.settings, state, setDateLocaleOpen]
+  );
+
+  const handleTimezoneSelect = React.useCallback(
+    (timezone: Timezone) => {
+      context.settings.setDatetimeFormat({
+        ...state,
+        timeZone: {
+          short: timezone.short,
+          full: timezone.full,
+        },
+      });
+    },
+    [context.settings, state]
+  );
+
+  const dateLocaleOptions = React.useMemo(
+    () =>
+      locales.map((locale) => (
+        <SelectOption
+          key={locale.key}
+          description={locale.key}
+          value={{
+            ...locale,
+            toString: () => locale.name,
+            compareTo: (val) => locale.name === val.name,
+          }}
+        >
+          {locale.name}
+        </SelectOption>
+      )),
+    []
+  );
+
+  const handleDateLocaleFilter = React.useCallback(
+    (_, value: string) => {
+      if (!value) {
+        return dateLocaleOptions;
+      }
+      const matchExp = new RegExp(value, 'i');
+      return dateLocaleOptions.filter(
+        (opt) => matchExp.test(opt.props.value.name) || matchExp.test(opt.props.description)
+      );
+    },
+    [dateLocaleOptions]
+  );
+
+  return (
+    <Stack hasGutter>
+      <StackItem key={'date-locale-select'}>
+        <FormGroup>
+          <HelperText>
+            <HelperTextItem>{'Select current date locale.'}</HelperTextItem>
+          </HelperText>
+          <Select
+            isOpen={dateLocaleOpen}
+            onToggle={handleDateToggle}
+            isFlipEnabled
+            menuAppendTo="parent"
+            selections={{
+              ...state.dateLocale,
+              toString: () => state.dateLocale.name,
+              compareTo: (val) => state.dateLocale.name === val.name,
+            }}
+            hasInlineFilter
+            maxHeight={'16em'}
+            onFilter={handleDateLocaleFilter}
+            onSelect={handleDateLocaleSelect}
+          >
+            {dateLocaleOptions}
+          </Select>
+        </FormGroup>
+      </StackItem>
+      <StackItem key={'timezone-select'}>
+        <FormGroup>
+          <HelperText>
+            <HelperTextItem>{'Select current timezone.'}</HelperTextItem>
```

Comment:
Should these messages be localized?

Reply:
ah yes I will start localization texts within the datetimepicker. But I will leave out any related to setting page tho (preparing another PR for that).